client/src/client.rs

```diff
@@ -534,6 +498,53 @@ impl Client {
     }
 }
 
+/// Client implements a JSON-RPC client for the Bitcoin Core daemon or compatible APIs.
+pub struct Client {
+    client: jsonrpc::client::Client,
+}
+
+impl Client {
+    /// Creates a client to a bitcoind JSON-RPC server.
+    pub fn new(url: String, user: Option<String>, pass: Option<String>) -> Self {
+        debug_assert!(pass.is_none() || user.is_some());
```

Comment:
❓  panics if there is no password or there is a user? I am not sure to understand why.

Reply:
No. **Does not panic** if there is no password or there is a user. This means that if you enter a password without a username, it panics.

Reply:
Ah, I see thanks.
Why using `debug_assert!` though? Can you elaborate?

Reply:
To be honest, I copied that like is from a prior client implementation. Don't even know which one.
Well, you never want panics in production, so it's more like a help against mistakes while testing.

I would guess the motivation of the original author was to want to avoid having to return a Result but also don't want to cause panics in production.

The server will give you an authentication error anyways if you run this in production.

Reply:
IMHO, I don't think it's a necessary check.
Either return a result and force the user to pass password and username (quite opinionated) or do not do checks :)